stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -75,68 +85,116 @@ public String getBenchDescription() {
         "$ bin/alluxio runClass alluxio.stress.cli.fuse.fuseIOBench --operation Read \\",
         "--local-path /mnt/alluxio-fuse/FuseIOTest --num-dirs 32 --num-files-per-dir 10 \\",
-        "--file-size 100m --threads 16 --warmup 5s --duration 30s",
+        "--file-size 100m --threads 16 --warmup 5s --duration 30s --cluster",
         ""
     ));
   }
 
   @Override
   public void prepare() throws Exception {
+    if (mBaseParameters.mCluster) {
+      // For cluster mode, this function is called once before the job is submitted to the job
+      // service. Nothing should be done, otherwise the bench will break.
+      return;
+    }
     if (mParameters.mThreads > mParameters.mNumDirs) {
       throw new IllegalArgumentException(String.format(
           "Some of the threads are not being used. Please set the number of directories to "
               + "be at least the number of threads, preferably a multiple of it."
       ));
     }
+    File localPath = new File(mParameters.mLocalPath);
+
     if (mParameters.mOperation == FuseIOOperation.WRITE) {
       LOG.warn("Cannot write repeatedly, so warmup is not possible. Setting warmup to 0s.");
       mParameters.mWarmup = "0s";
       for (int i = 0; i < mParameters.mNumDirs; i++) {
-        Files.createDirectories(Paths.get(mParameters.mLocalPath + "/" + i));
+        Files.createDirectories(Paths.get(String.format(
+            TEST_DIR_STRING_FORMAT, mParameters.mLocalPath, mBaseParameters.mId, i)));
       }
+      return;
+    }
+    if ((mParameters.mOperation == FuseIOOperation.REMOTE_READ
+        || mParameters.mOperation == FuseIOOperation.CLUSTER_READ)
+        && !mBaseParameters.mDistributed) {
+      throw new IllegalArgumentException(String.format(
+          "Single-node Fuse IO stress bench doesn't support RemoteRead or ClusterRead."
+      ));
+    }
+    // find 0-based id, and make sure directories and job workers are 1-to-1
+    File[] jobWorkerDirs = localPath.listFiles();
+    if (jobWorkerDirs == null) {
+      throw new IOException(String.format(
+          "--local-path %s is not a valid path for this bench. Make sure using the correct path",
+              mParameters.mLocalPath
+      ));
+    }
+    int numJobWorkers;
+    try (JobMasterClient client = JobMasterClient.Factory.create(
+        JobMasterClientContext.newBuilder(ClientContext.create(new InstancedConfiguration(
+            ConfigurationUtils.defaults()))).build())) {
+      numJobWorkers = client.getAllWorkerHealth().size();
+    }
+    if (numJobWorkers != jobWorkerDirs.length) {
+      throw new IllegalStateException("Some worker crashed or joined after data are written. "
+          + "The test is stopped.");
+    }
+    mJobWorkerDirNames = Arrays.asList(jobWorkerDirs).stream()
+        .map(file -> file.getName())
+        .collect(Collectors.toList());
+    try {
+      mJobWorkerZeroBasedId = mJobWorkerDirNames.indexOf(mBaseParameters.mId);
```

Comment:
For cluster mode, the time when it is called before figuring out it's cluster mode, it will enter https://github.com/Alluxio/alluxio/blob/5a14e865dda2248db30a692bc4056d4403db6045/stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java#L127-L129 and does nothing. The whole prepare() method is only useful when the job workers are executing.